jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP

```diff
@@ -34,7 +34,8 @@ RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx \
 
 # Add UFSDA to PYTHONPATH
 ufsdaPATH="${HOMEgfs}/sorc/gdas.cd/ush/"
-pyiodaPATH="${HOMEgfs}/sorc/gdas.cd/build/lib/python3.7/"
+py_ver=$(regex="[0-9]+\.[0-9]+"; if [[ $(python --version) =~ ${regex} ]]; then echo "${BASH_REMATCH[0]}"; fi)
+pyiodaPATH="${HOMEgfs}/sorc/gdas.cd/build/lib/python${py_ver}/"
```

Comment:
Instead of repeating ourselves, let's do this:

Add a new function to `ush/preamble.sh`:
```
detect_py_ver() {
    regex="[0-9]+\.[0-9]+"
    if [[ $(python --version) =~ ${regex} ]]; then
	    echo "${BASH_REMATCH[0]}"
    else
	    echo "FATAL ERROR: Could not detect the python version"
	    exit 1
    fi
}
```

Then these all become:
```suggestion
pyiodaPATH="${HOMEgfs}/sorc/gdas.cd/build/lib/python$(detect_py_ver)/"
```

Reply:
We're going to be moving the growing number of bash functions to a new location soon, but for now preamble is the correct place.

Reply:
Thank you very much @WalterKolczynski-NOAA for your guidance.  This is an excellent suggestion - much cleaner, compact, and centralized.  Suggested refactor committed to `RussTreadon-NOAA:feature/gdasapp_paths` at c365e992